source/slang/ir.h

```diff
@@ -959,18 +962,23 @@ struct IRStructField : IRInst
 // *not* contain the keys, because code needs to be able to
 // reference the keys from scopes outside of the struct.
 //
-struct IRStructType : IRInst
+struct IRStructType : IRType
```

Comment:
No. `IRStructType` used to inherit from `IRGlobalValue`, and then we had the whole song and dance of making an `as<IRType>(someInstThatIsAStruct)` work by arranging the ranges of the opcodes to line up just so.

When I removed `IRGlobalValue` I changed all code that referred to it to use `IRInst` instead (which included this inheritance clause), without looking at specific cases. There may be specific cases where eliminating `IRGlobalValue` means that:

* An IR instruction type can now inherit from a base that it couldn't before and clarify the hierarchy (this is the case for `IRStructType : IRType`)

* An IR instruction type can now have operands where it couldn't before (because `IRGlobalValue`/`IRParentInst` couldn't have operands). We haven't taken advantage of this yet, but `IRStructType` and `IRWitnessTable` are good candidates (we could eliminate `IRStructField` and `IRWitnessTableEntry` completely).